### PR TITLE
GH-235: As a developer I want a prepare def-use analyses by switching from path to branch semantics

### PR DESCRIPTION
**Pull request by mmews-n4**
see #235

### CODE REVIEW

File: plugins/org.eclipse.n4js.flowgraphs/src/org/eclipse/n4js/flowgraphs/analyses/EdgeGuideMerged.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ */
+package org.eclipse.n4js.flowgraphs.analyses;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ */
```
Comment (mor-n4) [inline plugins/org.eclipse.n4js.flowgraphs/src/org/eclipse/n4js/flowgraphs/analyses/EdgeGuideMerged.java]: documentation?

File: plugins/org.eclipse.n4js.flowgraphs/src/org/eclipse/n4js/flowgraphs/analyses/BranchWalker.java [hunk 2]
```diff
@@ -20,28 +21,37 @@
 import org.eclipse.n4js.flowgraphs.model.RepresentingNode;
 import org.eclipse.n4js.n4JS.ControlFlowElement;
 
-/** see {@link PathWalkerInternal} */
-abstract public class PathWalker extends PathWalkerInternal {
-	ControlFlowElement pLastCFE;
+/** see {@link BranchWalkerInternal} */
+abstract public class BranchWalker extends BranchWalkerInternal {
```
Comment (mor-n4) [inline plugins/org.eclipse.n4js.flowgraphs/src/org/eclipse/n4js/flowgraphs/analyses/BranchWalker.java]: Regarding the rename `PathWalker` -> `BranchWalker` (and the corresponding renames of other classes and newly added classes): did you update the design doc?

What about other adjustments of the explanations in the design doc?

Comment (mmews-n4) [reply plugins/org.eclipse.n4js.flowgraphs/src/org/eclipse/n4js/flowgraphs/analyses/BranchWalker.java]: Yes, I also updated the design doc (see other PR).

File: plugins/org.eclipse.n4js.flowgraphs/src/org/eclipse/n4js/flowgraphs/analyses/EdgeGuideMerged.java (hunk 1 above)
Comment (mmews-n4) [reply plugins/org.eclipse.n4js.flowgraphs/src/org/eclipse/n4js/flowgraphs/analyses/EdgeGuideMerged.java]: I will add that!

**Comment by mmews-n4** [comment]
Thank you!

**Pull request merged**